root/static/scripts/edit/URLCleanup.js

```diff
@@ -1223,6 +1223,28 @@ const CLEANUPS = {
       return url;
     }
   },
+  niconicouser: {
+    match: [new RegExp("^(https?://)?([^/]+\\.)?(nicovideo\\.jp)/user", "i")],
+    type: LINK_TYPES.socialnetwork,
+    clean: function(url) {
+      url = url.replace((/^https:\/\/www\.nicovideo\.jp\/user\/([0-9]+).*$/, "https://www.nicovideo.jp/user/$1");
```

Comment:
Pattern should be relaxed to match loose copy/paste missing scheme or subdomain. I suggest using `/^(?:https?:\/\/)?(?:[^\/]+\.)?` instead of `/^https:\/\/www\.`

Reply:
Should I keep www in the `https://www.nicovideo.jp/user/$1` ?

Reply:
Yes, this is the replacement string, and `niconico.jp` redirects to `ẁww.niconico.jp` which seems to be the canonical address for this website.